### PR TITLE
Moving prompt for NTP options to install_check

### PR DESCRIPTION
**Pull request by Tiboris**
In a interactive installation of freeipa server a promt asks for NTP related
options after install_check has been called. As it may cause confusion to users
moving to install_check methods where the prompt for other options is being done.

Resolves: https://pagure.io/freeipa/issue/7930

### CODE REVIEW

**Comment by xxblx** [comment]
@Tiboris tests fail with 
```
2019-05-02 09:45:27,627    DEBUG  transport.py               558 DEBUG    Error checking LDAP: Server is unwilling to perform: Minimum SSF not met.
```
Please do a rebase on https://github.com/freeipa/freeipa/commit/294aa3a33375dc246b2a733fce3cbd09a39071a0. That should fix PR CI failures.

File: ipaserver/install/server/install.py [hunk 1]
```diff
@@ -709,6 +714,13 @@ def install_check(installer):
                   "Directory unless\nthe realm name of the IPA server matches "
                   "its domain name.\n\n")
 
+    if options.ntp_servers or options.ntp_pool:
+        for server in options.ntp_servers:
+            print("NTP server:       %s" % server)
```
[the remark this answers is not in the document]
Comment (Tiboris) [reply ipaserver/install/server/install.py]: Sure, will do.

File: ipaserver/install/server/install.py [hunk 2]
```diff
@@ -427,13 +428,13 @@ def install_check(installer):
 
     if not options.no_ntp:
         try:
-            ipaclient.install.timeconf.check_timedate_services()
-        except ipaclient.install.timeconf.NTPConflictingService as e:
+            timeconf.check_timedate_services()
+        except timeconf.NTPConflictingService as e:
             print("WARNING: conflicting time&date synchronization service '{}'"
```
[the remark this answers is not in the document]
Comment (Tiboris) [reply ipaserver/install/server/install.py]: Okay!

File: ipaclient/install/client.py [hunk 3]
```diff
@@ -2472,12 +2486,9 @@ def sync_time(options, fstore, statestore):
 
     if not options.ntp_servers and not options.ntp_pool:
         ds = discovery.IPADiscovery()
-        ntp_servers = ds.ipadns_search_srv(cli_domain, '_ntp._udp',
-                                           None, break_on_first=False)
-        if not ntp_servers and not options.unattended:
-            options.ntp_servers, options.ntp_pool = timeconf.get_time_source()
-        else:
-            options.ntp_servers = ntp_servers
+        options.ntp_servers = ds.ipadns_search_srv(
```
Comment (rcritten) [inline ipaclient/install/client.py]: Is it bad form to update the options passed in vs those things detected? Could this make debugging harder later?

Comment (Tiboris) [reply ipaclient/install/client.py]: Hello @rcritten, 

yes I agree, actual behaviour could make debugging harder however i think approach described below is good and I will add another logging message and comments to the code itself.
```python
if installation is interactive and no ntp options had been passed:
    ask user for ntp options interactively
if no ntp options are passesd to installation:
    try autodiscovery for DNS record
if installation has no ntp options:
    skip configuration and use default one

synchronize time
```
Would that be acceptable alongside with update of design document?

Comment (rcritten) [reply ipaclient/install/client.py]: I agree with the steps and the purpose. I found it quite jarring to be prompted for NTP after answering yes to proceed with the installation, so I'm glad you're addressing this!
Is adding a new set of variables, ntp_server/ntp_pool so bad? It seems wrong to twiddle with options.*. We do this other places in the code as well, e.g. hostname, cli_domain, cli_server,...

Comment (Tiboris) [reply ipaclient/install/client.py]: Sorry I have misunderstood. Updating PR.

Comment (Tiboris) [reply ipaclient/install/client.py]: https://github.com/freeipa/freeipa/blob/254ba5240e897da04eea06b4f02dea579a9a6c54/ipaclient/install/client.py#L2476

File: ipatests/test_integration/test_external_ca.py [hunk 4]
```diff
@@ -83,7 +83,7 @@ def install_server_external_ca_step1(host, extra_args=()):
 def install_server_external_ca_step2(host, ipa_ca_cert, root_ca_cert,
                                      raiseonerr=True):
     """Step 2 to install the ipa server with external ca"""
-    args = ['ipa-server-install',
+    args = ['ipa-server-install', '-U', '-r', host.domain.realm,
             '-a', host.config.admin_password,
```
Comment (rcritten) [inline ipatests/test_integration/test_external_ca.py]: Is this change related?

Comment (Tiboris) [reply ipatests/test_integration/test_external_ca.py]: Yes @rcritten,

I found out that it is the external CA second step that was  causing a timeout for PRCI (see so many re-runs). Because of that I added an unattended option to this task. Without it it got stuck at new NTP prompt and then timeouted.

However it may happen that more nightly tests which rely on a fact that if all necessary options were passes an unattended option is not needed.

I see missing unattended option here as well because test is passing stdin text:

https://github.com/freeipa/freeipa/blob/master/ipatests/test_integration/test_installation.py#L587-L572

for this one a stdin input needs to be updated to have a ntp ptompt stdin text passed to installer.

**Comment by Tiboris** [comment]
master:

* e3f35843dc33fa6a3fb3071c5c89ecde9db1ce5e Moving prompt for NTP options to install_check
* be39d3a99bfd676cd1661af7e8b2553337b4e1c8 ipatests: Add Unattended option to external ca task
